### PR TITLE
[docs] use katex instead of mathjax

### LINKED ISSUE #1576: [docs] use katex instead of mathjax
## 🚀 Feature
Currently, math equation in the `contrib.metrics` are so small that we have to zoom to look at.
I want to suggest for using `katex` instead of `mathjax` and also `pytorch_sphinx_theme` has preloaded with `katex` font.

What's need to change here is
- install `sphinxcontrib-katex`
- add `sphinxcontrib.katex` in `extensions` of `conf.py`
- add the below code block in head tag of `layout.html`
```html
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css" integrity="sha384-AfEj0r4/OFrOo5t7NnNe46zW/tFgW6x/bCJG8FqQCEo3+Aro6EYUG4+cU+KJWu/X" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.js" integrity="sha384-g7c+Jr9ZivxKLnZTDUhnkOnsh30B4H0rpLUpJ4jAIKs4fnJI+sEnkvrMWph2EDg4" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/contrib/auto-render.min.js" integrity="sha384-mll67QQFJfxn0IYznZYonOWZ644AWYC+Pt2cHqMaRhXVrursRwvLnLaebdGIlYNa" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>
```
Katex: https://katex.org/docs/browser.html

### PR DESCRIPTION
**Pull request by ydcjeff**
Fixes #1576 

Description: use katex instead of mathjax for math equations as `pytorch_sphinx_theme` is loaded with `katex` fonts.

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by ydcjeff** [comment]
Seems like @vfdev-5 needs to update on netlify side.
```js
$ npm install katex
$ PATH="${PATH}:$(pwd)/node_modules/.bin"
```
https://sphinxcontrib-katex.readthedocs.io/en/0.7.1/installation.html

**Comment by ydcjeff** [comment]
For `build docs` workflow to work, we need to install katex

**Comment by vfdev-5** [comment]
> Seems like @vfdev-5 needs to update on netlify side.
> 
> ```js
> $ npm install katex
> $ PATH="${PATH}:$(pwd)/node_modules/.bin"
> ```
> 
> https://sphinxcontrib-katex.readthedocs.io/en/0.7.1/installation.html

Let me add that.

> For build docs workflow to work, we need to install katex in github actions

@ydcjeff here: https://github.com/pytorch/ignite/blob/master/.github/workflows/install_docs_deps.sh

**Comment by vfdev-5** [comment]
Looks good now : https://deploy-preview-1579--pytorch-ignite-preview.netlify.app/contrib/metrics.html#module-ignite.contrib.metrics.regression
👍

File: .github/workflows/install_docs_deps.sh [hunk 1]
```diff
@@ -1,3 +1,6 @@
+npm install katex
+PATH="${PATH}:$(pwd)/node_modules/.bin"
```
Comment (vfdev-5) [inline .github/workflows/install_docs_deps.sh:2]: I think env var setup wont work if we call it like `bash install_docs_deps.sh` and maybe it is not necessary if npm is already present in github actions...

Comment (ydcjeff) [reply .github/workflows/install_docs_deps.sh]: sudo mode on for `katex` 😁

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@ jobs:
         with:
           python-version: 3.7
 
+      - run: sudo npm install katex@0.12.0 -g
```
Comment (ydcjeff) [inline .github/workflows/docs.yml:18]: pin the version here as in the `layout.html`

Comment (vfdev-5) [reply .github/workflows/docs.yml:18]: do you know if there is a special tag like latest to install the latest version here and in the layout ?

Comment (ydcjeff) [reply .github/workflows/docs.yml:18]: TIL; you can use `latest` in the tag, didnt know that.
https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/
https://cdn.jsdelivr.net/npm/katex@latest/dist/

Same size

So, do you want to use latest version both in CI and layout.html?

Comment (vfdev-5) [reply .github/workflows/docs.yml:18]: yes, please

Comment (ydcjeff) [reply .github/workflows/docs.yml]: Done

**Review by vfdev-5 — APPROVED** [review]
Thanks @ydcjeff !

**Pull request merged**